### PR TITLE
[ir] Add more comments to CFG optimizations and analyses

### PR DESCRIPTION
**Pull request by xumingkuan**
Related issue = #2193

### CODE REVIEW

File: taichi/ir/control_flow_graph.cpp [hunk 1]
```diff
@@ -113,35 +113,13 @@ bool CFGNode::may_contain_variable(const std::unordered_set<Stmt *> &var_set,
...
-        continue;
-      }
-      if (!reach_kill_variable(data_source_ptr)) {
-        reach_gen.insert(stmt);
-        reach_kill.insert(data_source_ptr);
-      }
-    }
-  }
-}
-
```
Comment (xumingkuan) [inline taichi/ir/control_flow_graph.cpp:137]: Moved to L234.

File: taichi/ir/control_flow_graph.cpp [hunk 2]
```diff
@@ -429,43 +442,46 @@ bool CFGNode::dead_store_elimination(bool after_lower_access) {
...
-              global_load->ret_type = atomic->ret_type;
-              // Notice that we have a load here.
-              live_in_this_node.insert(atomic->dest);
-              live_load_in_this_node[atomic->dest] = global_load.get();
-              killed_in_this_node.erase(atomic->dest);
-              replace_with(i, std::move(global_load), true);
-              modified = true;
-              continue;
-            }
-          } else {
```
Comment (xumingkuan) [inline taichi/ir/control_flow_graph.cpp:464]: Moved to L451 so that the short if-clause appears first.

File: taichi/ir/control_flow_graph.cpp [hunk 3]
```diff
@@ -239,14 +224,39 @@ Stmt *CFGNode::get_store_forwarding_data(Stmt *var, int position) const {
...
   if (!result_visible) {
+    // The data is store-to-load forwardable but not visible at the place we
+    // are going to forward. We cannot forward it in this case.
     return nullptr;
   }
   return result;
 }
 
+void CFGNode::reaching_definition_analysis(bool after_lower_access) {
+  // Calculate reach_gen and reach_kill.
```
Comment (k-ye) [inline taichi/ir/control_flow_graph.cpp:235]:
```suggestion
  // Calculate |reach_gen| and |reach_kill|.
```

**Review by k-ye — APPROVED** [review]
Great thanks!

**Pull request merged**